src/runtime/model.cc

```diff
@@ -3192,6 +3192,21 @@ PerfMetrics FFModel::update_metrics_task(const Task *task,
   return all_metrics;
 }
 
+template <>
+std::tuple<> FFModel::get_input_shape(const std::tuple<> &) {
```

Comment:
Can we make this just standard function and not a method on `FFModel`? It doesn't seem to rely on any internal state of `FFModel`